pkg/validator/output.go

```diff
@@ -53,6 +53,26 @@ type AuditData struct {
 	Score                uint
 }
 
+// FilterResultsBySeverityLevel includes results according to the provided severity level:
+// 'danger' is the least verbose, 'warning' is medium verbosity, default behavior will
+// include all results, which currently also includes 'ignore'
+func (res AuditData) FilterResultsBySeverityLevel(severityLevel config.Severity) AuditData {
+	resCopy := res
+	resCopy.Results = []Result{}
+
+	filteredResults := funk.Map(res.Results, func(auditDataResult Result) Result {
+		return auditDataResult.filterResultsBySeverityLevel(severityLevel)
+	}).([]Result)
+
+	for _, result := range filteredResults {
+		if result.isNotEmpty() {
+			resCopy.Results = append(resCopy.Results, result)
+		}
+	}
```

Comment:
Man 😓 the structure of the polaris output makes this harder than it should be.

Probably not for this PR, but if we wanted to do other kinds of filtering we could probably create a generic `AuditData.filter` method that takes in a `func(Result) bool`, so we only have to define the weird "drill into each result" logic in one place.

Reply:
Yes that would be great, have some sort of filter function that takes a predicate